source/agora/test/MultiRoundConsensus.d

```diff
@@ -0,0 +1,163 @@
+        {
+            this.round_number = round_num;
+            uint64_t res = super.computeHashNode(slot_idx, prev, is_priority,
+                round_num, node_id);
+            return res;
+        }
+    }
+
+    static class CustomValidator : TestValidatorNode
+    {
+        /// Ctor
+        public this (Config config, Registry* reg, immutable(Block)[] blocks,
+            ulong txs_to_nominate)
+        {
+            super(config, reg, blocks, txs_to_nominate);
+        }
+
+        ///
+        protected override CustomNominator getNominator (
+            NetworkManager network, KeyPair key_pair, Ledger ledger,
+            TaskManager taskman)
+        {
+            return new CustomNominator(network, key_pair, ledger, taskman,
+                this.txs_to_nominate);
+        }
+    }
+
+    static class CustomAPIManager : TestAPIManager
+    {
+        ///
+        public this (immutable(Block)[] blocks, TestConf test_conf)
+        {
+            super(blocks, test_conf);
+        }
+
+        /// set base class
+        public override void createNewNode (Config conf, string file, int line)
+        {
+            if (this.nodes.length == 0)
+            {
+                auto api = RemoteAPI!TestAPI.spawn!CustomValidator(
+                    conf, &this.reg, this.blocks, this.test_conf.txs_to_nominate,
+                    conf.node.timeout);
+                this.reg.register(conf.node.address, api.tid());
+                this.nodes ~= NodePair(conf.node.address, api);
+            }
+            else
+                super.createNewNode(conf, file, line);
+        }
+    }
+
+    TestConf conf = {
+        timeout : 10.seconds,
+        validators : 3,
+        validator_cycle : 10,
+        quorum_threshold : 66
+    };
+
+    auto network = makeTestNetwork!CustomAPIManager(conf);
+    network.start();
+    scope(exit) network.shutdown();
+    scope(failure) network.printLogs();
+    network.waitForDiscovery();
+    auto nodes = network.clients;
+    auto validator = network.clients[0];
+
+    // Make three of three validators disable to respond
+    nodes[1].ctrl.sleep(10.seconds, true);
+    nodes[2].ctrl.sleep(10.seconds, true);
+
+    // Block 1 with multiple consensus rounds
+    auto txs = genesisSpendable().map!(txb => txb.sign()).array();
+    txs.each!(tx => validator.putTransaction(tx));
+
+    Thread.sleep(10.seconds);
+
+    network.expectBlock(Height(1), 5.seconds);
+    assert(CustomNominator.round_number > 1,
+        format("The validator's round number: %s. Expected: above %s",
+            CustomNominator.round_number, 1));
+
+    // Block 2 with a round
+    txs = txs.map!(tx => TxBuilder(tx).sign()).array();
+    txs.each!(tx => validator.putTransaction(tx));
+    network.expectBlock(Height(2), 5.seconds);
+    assert(CustomNominator.round_number == 1,
+        format("The validator's round number: %s. Expected: %s",
+            CustomNominator.round_number, 1));
+}
```

Comment:
```suggestion
}

```

There might be a setting in your text editor to always add a newline at the end.

Reply:
> There might be a setting in your text editor to always add a newline at the end.

Yep, I enabled the option of "Files: Insert Final Newline" at VS Code. Thanks.